dbms/src/Flash/tests/gtest_aggregation_executor.cpp

```diff
@@ -0,0 +1,309 @@
+                             /* columnInfos= */
+                             {{types_col_name[0], TiDB::TP::TypeLong},
+                              {types_col_name[1], TiDB::TP::TypeDecimal},
+                              {types_col_name[2], TiDB::TP::TypeTiny},
+                              {types_col_name[3], TiDB::TP::TypeShort},
+                              {types_col_name[4], TiDB::TP::TypeLong},
+                              {types_col_name[5], TiDB::TP::TypeLongLong},
+                              {types_col_name[6], TiDB::TP::TypeFloat},
+                              {types_col_name[7], TiDB::TP::TypeDouble},
+                              {types_col_name[8], TiDB::TP::TypeDate},
+                              {types_col_name[9], TiDB::TP::TypeDatetime},
+                              {types_col_name[10], TiDB::TP::TypeString}},
+                             /* columns= */
+                             {toNullableVec<Int32>(types_col_name[0], col_id),
+                              toNullableVec<Decimal32>(types_col_name[1], col_decimal),
+                              toNullableVec<Int8>(types_col_name[2], col_tinyint),
+                              toNullableVec<Int16>(types_col_name[3], col_smallint),
+                              toNullableVec<Int32>(types_col_name[4], col_int),
+                              toNullableVec<Int64>(types_col_name[5], col_bigint),
+                              toNullableVec<Float32>(types_col_name[6], col_float),
+                              toNullableVec<Float64>(types_col_name[7], col_double),
+                              toNullableVec<MyDate>(types_col_name[8], col_mydate),
+                              toNullableVec<MyDateTime>(types_col_name[9], col_mydatetime),
+                              toNullableVec<String>(types_col_name[10], col_string)});
+
+        /// Create table for tests of aggregation functions
+        context.addMockTable(/* name= */ {db_name, table_name},
+                             /* columnInfos= */
+                             {{col_name[0], TiDB::TP::TypeLong},
+                              {col_name[1], TiDB::TP::TypeString},
+                              {col_name[2], TiDB::TP::TypeString},
+                              {col_name[3], TiDB::TP::TypeDouble}},
+                             /* columns= */
+                             {toNullableVec<Int32>(col_name[0], col_age),
+                              toNullableVec<String>(col_name[1], col_gender),
+                              toNullableVec<String>(col_name[2], col_country),
+                              toNullableVec<Float64>(col_name[3], col_salary)});
+    }
+
+    std::shared_ptr<tipb::DAGRequest> buildDAGRequest(std::pair<String, String> src, MockAstVec agg_funcs, MockAstVec group_by_exprs, MockColumnNameVec proj)
+    {
+        /// We can filter the group by column with project operator.
+        /// project is applied to get single column for comparison
```

Comment:
More comments of why use project, like “partial agg output group by columns"

Reply:
> More comments of why use project, like “partial agg output group by columns"

Done